src/main/java/org/opentripplanner/model/calendar/openinghours/OHCalendar.java

```diff
@@ -7,28 +7,15 @@
 import java.time.ZonedDateTime;
 import java.time.temporal.ChronoUnit;
 import java.util.List;
-import java.util.Objects;
-import org.opentripplanner.util.lang.ToStringBuilder;
-
-public class OHCalendar implements Serializable {
-
-  private final ZoneId zoneId;
-  private final List<OpeningHours> openingHours;
-  private final LocalDate startOfCalendar;
-  private final LocalDate endOfCalendar;
-
-  public OHCalendar(
-    LocalDate startOfCalendar,
-    LocalDate endOfCalendar,
-    ZoneId zoneId,
-    List<OpeningHours> openingHours
-  ) {
-    this.startOfCalendar = startOfCalendar;
-    this.endOfCalendar = endOfCalendar;
-    this.zoneId = zoneId;
-    this.openingHours = openingHours;
-  }
-
+import java.util.stream.Collectors;
+
+public record OHCalendar(
```

Comment:
Should we use records instead of classes when there is additional logic methods in addition to just holding information? I don't know but I remember there has been previously discussion that something shouldn't be a record.

Reply:
Not sure if the rule is that records must not have any methods.

I was more of the impression that data transfer objects can be records but logic classes cannot. This type is on the border so lets debate that in the meeting.

Reply:
This is a domain model object - not a data-transfer-object (DTO). There are several small pitfalls with records:
  - encapsulation in general and especially for arrays and collections.
  - shallow eq/hc
  - In this case the existing toString had features that is removed, please add it back.
  - Ackward JavaDoc

As @optionsome we have discussed this before, and I do not want to spent my time repeating it. I want our main domain classes to develop into rich classes with behavior, well encapsulated. Records are difficult to use this way, and require trick to be applied. So, unless most OTP developers care and master this, I am tempted to just stay away from using records inside the model.

Reply:
I disagree but it's not important to me to argue over it. I've removed the commit and it's now a class again.